python/pyarrow/tests/test_parquet.py

```diff
@@ -2107,8 +2221,33 @@ def _filter_partition(df, part_keys):
     return df[predicate].drop(to_drop, axis=1)
 
 
-@pytest.mark.pandas
-def test_read_multiple_files(tempdir):
+def assert_table_equal(left, right, check_metadata=False):
+    if left.equals(right, check_metadata=check_metadata):
+        return
+    
+    if not left.schema.equals(right.schema):
+        raise AssertionError("Schema not equal\nLeft:\n{0}\nRight:\n{1}".format(left.schema, right.schema))
+        
+    if check_metadata:
+        if not left.schema.equals(right.schema, check_metadata=True):
```

Comment:
Is this necessary? If you've reached this point then the schemas must be equal (modulo metadata)

Reply:
I just put this function here temporarily for debugging, there is a JIRA about adding this properly -> https://issues.apache.org/jira/browse/ARROW-2647